src/gate_kernels.jl

```diff
@@ -311,14 +312,11 @@ function apply_gate!(
     g_00, g_10, g_01, g_11 = g_matrix
     Threads.@threads for chunk_index = 0:n_chunks-1
         # my_amps is the group of amplitude generators which this `Task` will process
```

Comment:
Need new explanation here

Reply:
Yep will add in a future "lots o' explanatory comments" PR